### PR TITLE
chore: adds gradle, snyk and circle steps

### PR DESCRIPTION
**Pull request by kotharironak**

### CODE REVIEW

**Review by laxmanchekka — DISMISSED** [review]
Looks good to me.
@ravisingal : Please take a look once.

File: .circleci/config.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  codecov: codecov/codecov@1
+  snyk: snyk/snyk@0.0.10
+
+executors:
+  gradle_docker:
+    docker:
+      - image: cimg/openjdk:11.0
+  helm:
+    docker:
+      - image: hypertrace/helm-gcs-packager:0.1.1
```
Comment (ravisingal) [inline .circleci/config.yaml:13]: use version `0.3.0`

File: .circleci/config.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: v1-dependencies-{{ checksum "/tmp/checksum.txt" }}
+jobs:
+  build:
+    executor: gradle_docker
+    steps:
+      - setup_build_environment
+      - setup_remote_docker
```
Comment (ravisingal) [inline .circleci/config.yaml:66]:
```suggestion
      - setup_remote_docker:
            version: 19.03.12
```

File: .circleci/config.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  publish:
+    executor: gradle_docker
+    steps:
+      - setup_build_environment
+      - setup_remote_docker
+      - gradle:
+          args: :tag -Prelease
+      - add_ssh_keys:
+          fingerprints:
+            - '89:21:66:64:b1:9e:5a:a6:7e:8c:67:f8:9a:3c:90:7b'
```
Comment (ravisingal) [inline .circleci/config.yaml:83]: This is new ssh key?

File: LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,202 @@
+
+                                 Apache License
```
Comment (ravisingal) [inline LICENSE.txt:2]: Thi service is Apache licensed or `Traceable Community License`?

File: .circleci/config.yaml (hunk 3 above)
Comment (kotharironak) [reply .circleci/config.yaml:83]: No.

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-all.zip
```
Comment (ravisingal) [inline gradle/wrapper/gradle-wrapper.properties:3]: update to version `6.6.1`

File: .circleci/config.yaml (hunk 3 above)
Comment (kotharironak) [reply .circleci/config.yaml]: Done

File: LICENSE.txt (hunk 4 above)
Comment (kotharironak) [reply LICENSE.txt:2]: It's going to be a mix - span-normalizer is under Apache, but hypertrace-trace-enricher is Community.

Comment (kotharironak) [reply LICENSE.txt:2]: Let me go with the community version at the root level, and the individual jobs will have accordingly.

File: gradle/wrapper/gradle-wrapper.properties (hunk 5 above)
Comment (kotharironak) [reply gradle/wrapper/gradle-wrapper.properties]: done

File: .circleci/config.yaml [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+          args: build dockerBuildImages
+      - gradle:
+          args: jacocoTestReport
+      - codecov/upload:
+          flags: unit
+  publish:
+    executor: gradle_docker
+    steps:
+      - setup_build_environment
+      - setup_remote_docker
```
Comment (kotharironak) [inline .circleci/config.yaml:79]:
```suggestion
      - setup_remote_docker
          version: 19.03.12
```

**Review by ravisingal — APPROVED** [review]

**Pull request merged**